test/Microsoft.Extensions.Http.Polly.Test/PolicyHttpMessageHandlerTest.cs

```diff
@@ -103,8 +97,65 @@ public class PolicyHttpMessageHandlerTest
             Assert.Same(expected, response);
         }
 
+        [Fact]
+        public async Task SendAsync_NoContextSet_CreatesNewContext()
+        {
+            // Arrange
+            var policy = Policy.TimeoutAsync<HttpResponseMessage>(TimeSpan.FromSeconds(10));
+            var handler = new TestPolicyHttpMessageHandler(policy);
+
+            Context context = null;
+            var expected = new HttpResponseMessage();
+            handler.OnSendAsync = (req, c, ct) =>
+            {
+                context = c;
+
+                Assert.NotNull(context);
+                Assert.Same(context, req.GetPollyContext());
+                return expected;
+            };
+
+            var request = new HttpRequestMessage();
+
+            // Act
+            var response = await handler.SendAsync(request, CancellationToken.None);
+
+            // Assert
+            Assert.Same(context, request.GetPollyContext()); // We don't clean up the context
+            Assert.Same(expected, response);
+        }
+
+        [Fact]
+        public async Task SendAsync_ExistingContext_ReusesContext()
+        {
+            // Arrange
+            var policy = Policy.TimeoutAsync<HttpResponseMessage>(TimeSpan.FromSeconds(10));
+            var handler = new TestPolicyHttpMessageHandler(policy);
+
+            var expected = new HttpResponseMessage();
+            handler.OnSendAsync = (req, c, ct) =>
+            {
+                Assert.NotNull(c);
+                Assert.Same(c, req.GetPollyContext());
+                return expected;
+            };
+
+            var request = new HttpRequestMessage();
+            var context = new Context(Guid.NewGuid().ToString());
+            request.SetPollyContext(context);
+
+            // Act
+            var response = await handler.SendAsync(request, CancellationToken.None);
+
+            // Assert
```

Comment:
`Assert.AreSame(context, contextPassedToSendAsync);`

Reply:
This comment is a bit stale now, but I addressed the underlying problem